src/main/java/bdv/fx/viewer/render/RenderUnit.java

```diff
@@ -339,7 +346,16 @@ public void paint() {
 					null
 			);
 
-			renderTarget.drawOverlays(image::set);
+			if (renderedScreenScaleIndex != -1)
+			{
+				final Double renderedScreenScale;
+				synchronized (RenderUnit.this)
+				{
+					renderedScreenScale = renderedScreenScaleIndex >= 0 && renderedScreenScaleIndex < screenScales.length ? screenScales[renderedScreenScaleIndex] : null;
```

Comment:
Is the `>= 0` check necessary? We already checked `renderedScreenScaleIndex != -1`